### ISSUE TITLE
[Support]: Unable to update/install magit, magit-section, transient, with-editor, ghub, borg, epkg

### ISSUE THREAD

**Issue by deejayem** [progfolio/elpaca#343]
### Confirmation

- [X] I have checked the documentation (README, Wiki, docstrings, etc)
- [ ] I am  checking these without reading them.
- [X] I have searched previous issues to see if my question is a duplicate.

### Elpaca Version

Elpaca edc0dc2 grafted, HEAD -> master, origin/master, origin/HEAD                                                      
installer:      0.7
emacs-version:  GNU Emacs 29.1 (build 1, aarch64-apple-darwin23.5.0, Carbon Version 170 AppKit 2487.6)
git --version:  git version 2.45.2

### Operating System

Darwin Kernel Version 23.6.0: Mon Jul 29 21:14:30 PDT 2024; root:xnu-10063.141.2~1/RELEASE_ARM64_T6030 arm64 arm Darwin

### Description

1) Remove ~/.emacs.d/elpaca
2) Start Emacs, and everything works fine
3) Wait a day
4) Run elpaca-fetch-all, then mark magit (and everything else) to be merged (m), and execute it (x).

It fails to merge, because of the git error shown below. The same happens for some other packages as well, such as ghub, forge and with-editor.  I notice that these are all trying to use the melpa branch, but on a different system they're on main. I don't think there's anything interesting in my magit configuration, but you can see it here if you need to: https://codeberg.org/djm/dotfiles/src/branch/master/.emacs.d/lisp/init-git.el#L73

```
magit [MELPA|NonGNU-devel ELPA]
A Git porcelain inside Emacs

source: MELPA
url: https://github.com/magit/magit
menu item recipe:
( :package "magit"
  :fetcher github
  :repo "magit/magit"
  :branch "melpa"
  :files ("lisp/magit*.el"
          "lisp/git-rebase.el" "docs/magit.texi"
          "docs/AUTHORS.md" "LICENSE"
          "magit-pkg.el" (:exclude
                          "lisp/magit-section.el"))
  :source "MELPA")
full recipe:
( :package "magit"
  ;; Inherited from elpaca-order-functions.
  :depth 1
  :inherit t
  :protocol https
  ;; Inherited from elpaca-menu-item.
  :source "MELPA"
  :files ("lisp/magit*.el"
          "lisp/git-rebase.el" "docs/magit.texi"
          "docs/AUTHORS.md" "LICENSE"
          "magit-pkg.el" (:exclude
                          "lisp/magit-section.el"))
  :branch "melpa"
  :repo "magit/magit"
  :fetcher github)
dependencies:
  emacs         >= 26.1
  compat        >= 30.0.0.0
  dash          >= 2.19.1
  git-commit    >= 4.0.0
  magit-section >= 4.0.0
  seq           >= 2.24
  transient     >= 0.7.4
  with-editor   >= 3.4.1
dependents: 
  forge
  magit-delta
installed version: 4.0.0 e9ebf8ec
statuses:
  (failed reclone merging queued)
files:
  /magit/LICENSE                       → /magit/LICENSE
  /magit/docs/AUTHORS.md               → /magit/AUTHORS.md
  /magit/docs/magit.texi               → /magit/magit.texi
  /magit/lisp/git-rebase.el            → /magit/git-rebase.el
  /magit/lisp/magit-apply.el           → /magit/magit-apply.el
  /magit/lisp/magit-autorevert.el      → /magit/magit-autorevert.el
  /magit/lisp/magit-base.el            → /magit/magit-base.el
  /magit/lisp/magit-bisect.el          → /magit/magit-bisect.el
  /magit/lisp/magit-blame.el           → /magit/magit-blame.el
  /magit/lisp/magit-bookmark.el        → /magit/magit-bookmark.el
  /magit/lisp/magit-branch.el          → /magit/magit-branch.el
  /magit/lisp/magit-bundle.el          → /magit/magit-bundle.el
  /magit/lisp/magit-clone.el           → /magit/magit-clone.el
  /magit/lisp/magit-commit.el          → /magit/magit-commit.el
  /magit/lisp/magit-core.el            → /magit/magit-core.el
  /magit/lisp/magit-diff.el            → /magit/magit-diff.el
  /magit/lisp/magit-ediff.el           → /magit/magit-ediff.el
  /magit/lisp/magit-extras.el          → /magit/magit-extras.el
  /magit/lisp/magit-fetch.el           → /magit/magit-fetch.el
  /magit/lisp/magit-files.el           → /magit/magit-files.el
  /magit/lisp/magit-git.el             → /magit/magit-git.el
  /magit/lisp/magit-gitignore.el       → /magit/magit-gitignore.el
  /magit/lisp/magit-log.el             → /magit/magit-log.el
  /magit/lisp/magit-margin.el          → /magit/magit-margin.el
  /magit/lisp/magit-merge.el           → /magit/magit-merge.el
  /magit/lisp/magit-mode.el            → /magit/magit-mode.el
  /magit/lisp/magit-notes.el           → /magit/magit-notes.el
  /magit/lisp/magit-patch.el           → /magit/magit-patch.el
  /magit/lisp/magit-process.el         → /magit/magit-process.el
  /magit/lisp/magit-pull.el            → /magit/magit-pull.el
  /magit/lisp/magit-push.el            → /magit/magit-push.el
  /magit/lisp/magit-reflog.el          → /magit/magit-reflog.el
  /magit/lisp/magit-refs.el            → /magit/magit-refs.el
  /magit/lisp/magit-remote.el          → /magit/magit-remote.el
  /magit/lisp/magit-repos.el           → /magit/magit-repos.el
  /magit/lisp/magit-reset.el           → /magit/magit-reset.el
  /magit/lisp/magit-sequence.el        → /magit/magit-sequence.el
  /magit/lisp/magit-sparse-checkout.el → /magit/magit-sparse-checkout.el
  /magit/lisp/magit-stash.el           → /magit/magit-stash.el
  /magit/lisp/magit-status.el          → /magit/magit-status.el
  /magit/lisp/magit-submodule.el       → /magit/magit-submodule.el
  /magit/lisp/magit-subtree.el         → /magit/magit-subtree.el
  /magit/lisp/magit-tag.el             → /magit/magit-tag.el
  /magit/lisp/magit-transient.el       → /magit/magit-transient.el
  /magit/lisp/magit-wip.el             → /magit/magit-wip.el
  /magit/lisp/magit-worktree.el        → /magit/magit-worktree.el
  /magit/lisp/magit.el                 → /magit/magit.el
log:
  [2024-08-22 07:29:34] Package queued
  [2024-08-22 07:29:37] Continued by: elpaca--process
  [2024-08-22 07:29:37] Queueing Dependencies
  [2024-08-22 07:29:37] Blocked by: (with-editor magit-section git-commit)
  [2024-08-22 07:29:37] Blocked by: (magit-section git-commit)
  [2024-08-22 07:29:37] Blocked by: (magit-section git-commit)
  [2024-08-22 07:29:37] Blocked by: (git-commit)
  [2024-08-22 07:29:37] Blocked by: (git-commit)
  [2024-08-22 07:29:37] Continued by: elpaca--check-status
  [2024-08-22 07:29:37] Unblocked by: git-commit
  [2024-08-22 07:29:37] No Info dir file found
  [2024-08-22 07:29:37] Continued by: elpaca--add-info-path
  [2024-08-22 07:29:37] Activating package
  [2024-08-22 07:29:37] Package build dir added to load-path
  [2024-08-22 07:29:37] Caching autoloads
  [2024-08-22 07:29:37] Autoloads cached
  [2024-08-22 07:29:37] Continued by: elpaca--activate-package
  [2024-08-22 07:29:37] ✓ 2.642 secs
  [2024-08-22 07:30:00] Continued by: elpaca--process
  [2024-08-22 07:30:00] elpaca-queue-limit exceeded
  [2024-08-22 07:30:00] Continued by: elpaca--check-status
  [2024-08-22 07:30:00] Unblocked by: git-commit
  [2024-08-22 07:30:00] elpaca-queue-limit exceeded
  [2024-08-22 07:30:00] Continued by: elpaca--check-status
  [2024-08-22 07:30:00] Unblocked by: magit-section
  [2024-08-22 07:30:00] elpaca-queue-limit exceeded
  [2024-08-22 07:30:00] Continued by: elpaca--check-status
  [2024-08-22 07:30:00] Unblocked by: with-editor
  [2024-08-22 07:30:00] elpaca-queue-limit exceeded
  [2024-08-22 07:30:01] Continued by: elpaca--check-status
  [2024-08-22 07:30:01] Unblocked by: with-editor
  [2024-08-22 07:30:01] elpaca-queue-limit exceeded
  [2024-08-22 07:30:01] Continued by: elpaca--check-status
  [2024-08-22 07:30:01] Unblocked by: with-editor
  [2024-08-22 07:30:01] elpaca-queue-limit exceeded
  [2024-08-22 07:30:01] Continued by: elpaca--check-status
  [2024-08-22 07:30:01] Unblocked by: with-editor
  [2024-08-22 07:30:01] Fetching remotes
  [2024-08-22 07:30:01]  fetch --all -v
  [2024-08-22 07:30:02]   POST git-upload-pack (409 bytes)
  [2024-08-22 07:30:02]   POST git-upload-pack (275 bytes)
  [2024-08-22 07:30:05]   From https://github.com/magit/magit
  [2024-08-22 07:30:05]    + e9ebf8ec...622c6c37 melpa      -> origin/melpa  (forced update)
  [2024-08-22 07:30:05]    * [new tag]           0.5.1      -> 0.5.1
  [2024-08-22 07:30:05]    * [new tag]           0.6.1      -> 0.6.1
  [2024-08-22 07:30:05]    * [new tag]           0.7.0      -> 0.7.0
  [2024-08-22 07:30:05]    * [new tag]           0.8.0      -> 0.8.0
  [2024-08-22 07:30:05]    * [new tag]           0.8.1      -> 0.8.1
  [2024-08-22 07:30:05]    * [new tag]           0.8.2      -> 0.8.2
  [2024-08-22 07:30:05]    * [new tag]           1.0.0      -> 1.0.0
  [2024-08-22 07:30:05]    * [new tag]           1.1.0      -> 1.1.0
  [2024-08-22 07:30:05]    * [new tag]           1.1.1      -> 1.1.1
  [2024-08-22 07:30:05]    * [new tag]           1.1.2      -> 1.1.2
  [2024-08-22 07:30:05]    * [new tag]           1.2.0      -> 1.2.0
  [2024-08-22 07:30:05]    * [new tag]           1.4.0      -> 1.4.0
  [2024-08-22 07:30:05]    * [new tag]           1.4.1      -> 1.4.1
  [2024-08-22 07:30:05]    * [new tag]           1.4.2      -> 1.4.2
  [2024-08-22 07:30:05]    * [new tag]           2.1.0      -> 2.1.0
  [2024-08-22 07:30:05]    * [new tag]           2.10.0     -> 2.10.0
  [2024-08-22 07:30:05]    * [new tag]           2.10.1     -> 2.10.1
  [2024-08-22 07:30:05]    * [new tag]           2.10.2     -> 2.10.2
  [2024-08-22 07:30:05]    * [new tag]           2.10.3     -> 2.10.3
  [2024-08-22 07:30:05]    * [new tag]           2.11.0     -> 2.11.0
  [2024-08-22 07:30:05]    * [new tag]           2.12.0     -> 2.12.0
  [2024-08-22 07:30:05]    * [new tag]           2.12.1     -> 2.12.1
  [2024-08-22 07:30:05]    * [new tag]           2.13.0     -> 2.13.0
  [2024-08-22 07:30:05]    * [new tag]           2.13.1     -> 2.13.1
  [2024-08-22 07:30:05]    * [new tag]           2.2.0      -> 2.2.0
  [2024-08-22 07:30:05]    * [new tag]           2.2.1      -> 2.2.1
  [2024-08-22 07:30:05]    * [new tag]           2.2.2      -> 2.2.2
  [2024-08-22 07:30:05]    * [new tag]           2.3.0      -> 2.3.0
  [2024-08-22 07:30:05]    * [new tag]           2.3.1      -> 2.3.1
  [2024-08-22 07:30:05]    * [new tag]           2.4.0      -> 2.4.0
  [2024-08-22 07:30:05]    * [new tag]           2.4.1      -> 2.4.1
  [2024-08-22 07:30:05]    * [new tag]           2.5.0      -> 2.5.0
  [2024-08-22 07:30:05]    * [new tag]           2.6.0      -> 2.6.0
  [2024-08-22 07:30:05]    * [new tag]           2.6.1      -> 2.6.1
  [2024-08-22 07:30:05]    * [new tag]           2.6.2      -> 2.6.2
  [2024-08-22 07:30:05]    * [new tag]           2.7.0      -> 2.7.0
  [2024-08-22 07:30:05]    * [new tag]           2.8.0      -> 2.8.0
  [2024-08-22 07:30:05]    * [new tag]           2.9.0      -> 2.9.0
  [2024-08-22 07:30:05]    * [new tag]           v2.90.0    -> v2.90.0
  [2024-08-22 07:30:05]    * [new tag]           v2.90.1    -> v2.90.1
  [2024-08-22 07:30:05]    * [new tag]           v3.0.0     -> v3.0.0
  [2024-08-22 07:30:05]    * [new tag]           v3.1.0     -> v3.1.0
  [2024-08-22 07:30:05]    * [new tag]           v3.1.1     -> v3.1.1
  [2024-08-22 07:30:05]    * [new tag]           v3.2.0     -> v3.2.0
  [2024-08-22 07:30:05]    * [new tag]           v3.2.1     -> v3.2.1
  [2024-08-22 07:30:05]    * [new tag]           v3.3.0     -> v3.3.0
  [2024-08-22 07:30:05]    * [new tag]           v4.0.0     -> v4.0.0
  [2024-08-22 07:30:05] Continued by: #[128 "\302\301\303\300\"\"\207" [("Remotes fetched" nil) elpaca--process-sentinel apply append] 6 "

(fn &rest ARGS2)"]
  [2024-08-22 07:30:05] Remotes fetched
  [2024-08-22 07:30:05]  --no-pager log --reverse --since=2024-08-21 01:58:10 +0000 --pretty=%h %s (%ch) ..@{u}
  [2024-08-22 07:30:05]   cd959c78 magit-tag-release: Increase consistency and separation of concerns (8 hours ago)
  [2024-08-22 07:30:05]   622c6c37 Publish for Melpa (always "main" + one commit) (Wed 22:15 +0000)
  [2024-08-22 07:30:05] Continued by: #[128 "\302\301\303\300\"\"\207" [(nil nil) elpaca--process-sentinel apply append] 6 "

(fn &rest ARGS2)"]
  [2024-08-22 07:30:05] ✓ 7.728 secs
  [2024-08-22 07:30:28] Continued by: elpaca--process
  [2024-08-22 07:30:28]  merge --ff-only
  [2024-08-22 07:30:28] Merging updates
  [2024-08-22 07:30:28]   fatal: refusing to merge unrelated histories
  [2024-08-22 07:30:28] Merge failed
```

**Comment by progfolio** [comment]
Thanks for taking the time to fill out a bug report.

My advice is to update your declarations to use the main branch for those packages. e.g.

```emacs-lisp
(elpaca (magit :branch "main"))
```

@tarsius updated the MELPA recipes for several of  his packages to 
point to a CI-generated "melpa" branch, which includes a generated manual for MELPA builds. I'm unaware of any other advantage of this branch.  Elpaca should build the manual on its own from the main branch just fine.

**Comment by deejayem** [comment]
@progfolio thank you for the response. Setting the branch resolves the issue.

It seems more like a workaround than a fix though - I think most people would expect to be able to add `(use-package magit... ` to their config, and have everything work smoothly, without needing to specify the branch. So I will leave it up to you as to whether you want to close this, or if there's something that can be changed to improve things.

**Comment by bodnarlajos** [comment]
I have the same issue on win10.
![elpaca_issue](https://github.com/user-attachments/assets/6d159a20-a4e2-4db8-b26f-7f61992da6a1)

**Comment by progfolio** [comment]
> @progfolio thank you for the response. Setting the branch resolves the issue.

Glad to hear it.

> It seems more like a workaround than a fix though - I think most people would expect to be able to add `(use-package magit... ` to their config, and have everything work smoothly, without needing to specify the branch. So I will leave it up to you as to whether you want to close this, or if there's something that can be changed to improve things.

I can't control the recipes that are in the wild.
The only improvement would be for someone to create a menu which implements the altered recipes.
I'd prefer not to be on the hook for that, because it often involves packages which I don't use myself.
On the other hand, I usually end up being the one to do the research into how to fix the recipe when people ask for support here, so I may end up doing that eventually.

**Comment by progfolio** [comment]
> I have the same issue on win10. ![elpaca_issue](https://private-user-images.githubusercontent.com/41907523/360897348-6d159a20-a4e2-4db8-b26f-7f61992da6a1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0NTc3NDIsIm5iZiI6MTcyNDQ1NzQ0MiwicGF0aCI6Ii80MTkwNzUyMy8zNjA4OTczNDgtNmQxNTlhMjAtYTRlMi00ZGI4LWIyNmYtN2Y2MTk5MmRhNmExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDIzNTcyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyODdlYTYyYzUzZTAzOGM4ZjYzZmMyNWVkZWFjOGY2ZmQ0YzI1N2YxYjc3N2VmNDZiOGNkNzljNDUxYzkzZjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cqC-JP2_--EtiZFT4DNbNFgxIjOg4SS7Pzj3PHFSEvY)
> 

Please share text instead of screenshots in the future.

> What is wrong with my config and how I can solve it ?

The error in the log means that the version of transient you have installed is not new enough to work with the version of git-commit you have installed (which is needed for magit).
You can either update Emacs to get a newer version of  the built-in transient.el or install transient via Elpaca. 
e.g. `(elpaca (transient :branch "main"))`

If that does not work for you, please fill out a separate issue ticket and I can work with you there.

**Comment by tarsius** [comment]
> On the other hand, I usually end up being the one to do the research into how to fix the recipe when people ask for support here, so I may end up doing that eventually.

I'm certainly willing to work with you on finding a solution. It's bed time now, but I'll come back here tomorrow evening.

**Comment by bodnarlajos** [comment]
I will try to install transient on Monday. If it wont solve the problem, I will let you know! 
Thanks in advance!

**Comment by tarsius** [comment]
Initially I was going to suggest that elpaca and straight should ignore `:branch` if the value is `"melpa"`. That would save users from having to add `:branch "main"`.

But with that approach at least two issues remain. Users would still have to add `:pre-build ("make" "info")` and make and texinfo would have to be installed locally. @hlissner mentioned at https://github.com/doomemacs/doomemacs/issues/8007#issuecomment-2306537275 that he wouldn't want that.

So I think it is better after all, if elpaca and straight continued to use the branch that is set in melpa's recipe.

----

As far as I understand, elpaca and straight currently essentially fetch the latest version/snapshot/commit using

```
git pull --ff-only || signal error
```

Have you considered using the same approach as melpa?:


```
git fetch origin
git reset --hard origin/the-branch
```

This wouldn't just help in this case. Minor history rewrites happen all the time, e.g., "amend to HEAD to fix a typo" is quite common.

For the emacsmirror I used to use the same approach as elpaca and straight currently use. Every time someone rewrote history and it happened to pull before *and* after the rewrite, the update would fail, and I would have to go look at the situation and resolve it manually.

That meant that I had to look at very minor and inconsequential rewrites all the time. To avoid that, I eventually switched to an approach similar to that used by melpa.

**Comment by bodnarlajos** [comment]
I switched back to the package.el and I will try the elpaca later ...
Thanks for yours time!

**Comment by tarsius** [comment]
> Have you considered using the same approach as melpa? [...]

@progfolio Have you seen this suggestion and does it sound like a workable solution?

**Comment by progfolio** [comment]
> Have you considered using the same approach as melpa?

I have. If I go that route, I will have to implement a check which fails the order if the user has local changes prior to the hard reset. There will also be some minor refactoring needed with recipe resolution. I will experiment with it when I have time.

**Comment by tarsius** [comment]
Great, thanks!

**Comment by tarsius** [comment]
I'm checking the generated `*.texi` into `main` branches again.

I think my suggestion above makes still makes sense.

**Comment by meedstrom** [comment]
Just chiming in to agree with not treating "melpa" branch specially.  I also use a branch with that name, [for a different reason](https://github.com/melpa/melpa/pull/9132#issuecomment-2308520216).

**Comment by progfolio** [comment]
> Just chiming in to agree with not treating "melpa" branch specially. I also use a branch with that name, [for a different reason](https://github.com/melpa/melpa/pull/9132#issuecomment-2308520216).

I have no plans of adding any branch-name based heuristics for Elpaca's behavior, so nothing to worry about there.

**Comment by domsch1988** [comment]
Even with all sub-packages pinned to the main branch, i'm not able to get magit installed anymore. My Config for that now looks like this:
```
(elpaca (magit :branch "main" :pre-build ("make" "info")))
(elpaca (forge :branch "main"))
(elpaca (ghub :branch "main"))
(elpaca (transient :branch "main"))
(elpaca (with-editor :branch "main"))
```
But transient and with-editor still fail to build
**with-editor**
```
source: MELPA
url: https://github.com/magit/with-editor
menu item recipe:
( :package "with-editor"
  :fetcher github
  :repo "magit/with-editor"
  :branch "melpa"
  :files ("*.el" "*.el.in" "dir" "*.info"
	  "*.texi" "*.texinfo" "doc/dir"
	  "doc/*.info" "doc/*.texi"
	  "doc/*.texinfo" "lisp/*.el" (:exclude
				       ".dir-locals.el" "test.el" "tests.el"
				       "*-test.el" "*-tests.el" "LICENSE"
				       "README*" "*-pkg.el"))
  :source "MELPA")
full recipe:
( :package "with-editor"
  ;; Inherited from declaration.
  :branch "main"
  ;; Inherited from elpaca-order-functions.
  :depth 1
  :inherit t
  :protocol https
  ;; Inherited from elpaca-menu-item.
  :source "MELPA"
  :files (:defaults)
  :repo "magit/with-editor"
  :fetcher github)
dependencies:
  emacs  >= 26.1
  compat >= 30.0.0.0
dependents: 
  embark-vc
  forge
  magit
  treemacs-magit
installed version: 3.4.1 5dded4e
statuses:
  (failed reclone fetching-remotes queued)
files:
  $REPOS/with-editor/lisp/with-editor.el → $BUILDS/with-editor/with-editor.el
log:
  [2024-09-03 08:58:45] Package queued
  [2024-09-03 08:58:45] Continued by: elpaca--process
  [2024-09-03 08:58:45] Queueing Dependencies
  [2024-09-03 08:58:45] Continued by: elpaca--queue-dependencies
  [2024-09-03 08:58:45] No Info dir file found
  [2024-09-03 08:58:45] Continued by: elpaca--add-info-path
  [2024-09-03 08:58:45] Activating package
  [2024-09-03 08:58:45] Package build dir added to load-path
  [2024-09-03 08:58:45] Caching autoloads
  [2024-09-03 08:58:45] Autoloads cached
  [2024-09-03 08:58:45] Continued by: elpaca--activate-package
  [2024-09-03 08:58:45] ✓ 0.018 secs
  [2024-09-03 08:58:54] Updating
  [2024-09-03 08:58:55] Continued by: elpaca--process
  [2024-09-03 08:58:55] Fetching remotes
  [2024-09-03 08:58:55] $git fetch --all -v
  [2024-09-03 08:58:55]   POST git-upload-pack (409 bytes)
  [2024-09-03 08:58:55]   Schwerwiegend: Konnte Remote-Referenz refs/heads/melpa nicht finden.
  [2024-09-03 08:58:55] Subprocess error (see previous log entries)
  [2024-09-03 08:59:02] Continued by: elpaca--process
  [2024-09-03 08:59:02] Fetching remotes
  [2024-09-03 08:59:02] $git fetch --all -v
  [2024-09-03 08:59:02]   POST git-upload-pack (409 bytes)
  [2024-09-03 08:59:02]   Schwerwiegend: Konnte Remote-Referenz refs/heads/melpa nicht finden.
  [2024-09-03 08:59:02] Subprocess error (see previous log entries)
```
**transient**
```
source: MELPA
url: https://github.com/magit/transient
menu item recipe:
( :package "transient"
  :fetcher github
  :repo "magit/transient"
  :branch "melpa"
  :files ("*.el" "*.el.in" "dir" "*.info"
	  "*.texi" "*.texinfo" "doc/dir"
	  "doc/*.info" "doc/*.texi"
	  "doc/*.texinfo" "lisp/*.el" (:exclude
				       ".dir-locals.el" "test.el" "tests.el"
				       "*-test.el" "*-tests.el" "LICENSE"
				       "README*" "*-pkg.el"))
  :source "MELPA")
full recipe:
( :package "transient"
  ;; Inherited from declaration.
  :pre-build ("make" "info")
  :branch "main"
  ;; Inherited from elpaca-order-functions.
  :depth 1
  :inherit t
  :protocol https
  ;; Inherited from elpaca-menu-item.
  :source "MELPA"
  :files (:defaults)
  :repo "magit/transient"
  :fetcher github)
dependencies:
  emacs  >= 26.1
  compat >= 30.0.0.0
  seq    >= 2.24
dependents: 
  embark-vc
  forge
  magit
  treemacs-magit
installed version: 0.7.4 47960ef
statuses:
  (failed reclone fetching-remotes queued)
files:
  $REPOS/transient/lisp/transient.el → $BUILDS/transient/transient.el
log:
  [2024-09-03 08:58:45] Package queued
  [2024-09-03 08:58:45] Continued by: elpaca--process
  [2024-09-03 08:58:45] Queueing Dependencies
  [2024-09-03 08:58:45] Continued by: elpaca--check-status
  [2024-09-03 08:58:45] Unblocked by: compat
  [2024-09-03 08:58:45] No Info dir file found
  [2024-09-03 08:58:45] Continued by: elpaca--add-info-path
  [2024-09-03 08:58:45] Activating package
  [2024-09-03 08:58:45] Package build dir added to load-path
  [2024-09-03 08:58:45] Caching autoloads
  [2024-09-03 08:58:45] Autoloads cached
  [2024-09-03 08:58:45] Continued by: elpaca--activate-package
  [2024-09-03 08:58:45] ✓ 0.017 secs
  [2024-09-03 08:58:54] Updating
  [2024-09-03 08:58:55] Continued by: elpaca--process
  [2024-09-03 08:58:55] Fetching remotes
  [2024-09-03 08:58:55] $git fetch --all -v
  [2024-09-03 08:58:55]   POST git-upload-pack (409 bytes)
  [2024-09-03 08:58:55]   Schwerwiegend: Konnte Remote-Referenz refs/heads/melpa nicht finden.
  [2024-09-03 08:58:55] Subprocess error (see previous log entries)
  [2024-09-03 09:02:30] Continued by: elpaca--process
  [2024-09-03 09:02:30] Fetching remotes
  [2024-09-03 09:02:30] $git fetch --all -v
  [2024-09-03 09:02:31]   POST git-upload-pack (409 bytes)
  [2024-09-03 09:02:31]   Schwerwiegend: Konnte Remote-Referenz refs/heads/melpa nicht finden.
  [2024-09-03 09:02:31] Subprocess error (see previous log entries)
```

I also tried with just magit pinned or pulling the dependencies through use-package, both don't seem to work for me.

**Comment by progfolio** [comment]
> Edit: Forget about this. I deleted my elpaca Directory and started "fresh". Now
> it works. I'll leave this info here for those searching.

Please see: https://github.com/progfolio/elpaca/wiki/Troubleshooting#dont-delete-everything
  
> Even with all sub-packages pinned to the main branch, i'm not able to get magit installed anymore. My Config for that now looks like this:
> ```
> (elpaca (magit :branch "main" :pre-build ("make" "info")))
> (elpaca (forge :branch "main"))
> (elpaca (ghub :branch "main"))
> (elpaca (transient :branch "main"))
> (elpaca (with-editor :branch "main"))
> ```

This should no longer be necessary. The MELPA recipes have been reverted to track the main branch again:

https://github.com/melpa/melpa/commit/a74629656e9a23133219a0bd805982f1497b35d7

What will be necessary is to `M-x elpaca-update-menus` for the MELPA menu so those updated recipes can be inherited from.

>   [2024-09-03 08:58:55] $git fetch --all -v
>   [2024-09-03 08:58:55]   POST git-upload-pack (409 bytes)
>   [2024-09-03 08:58:55]   Schwerwiegend: Konnte Remote-Referenz refs/heads/melpa nicht finden.

This is to be expected as the `melpa` branch no longer exists in those repositories.

The `git-commit` package is also distributed as part of magit now:
https://github.com/melpa/melpa/commit/301c815541da7eaa552d834bfbed3e6633bd4136

So the best course of action would be:

0. remove the explicit declarations for magit's dependencies from your init file (unless you need them for other customizations).
1. `M-x elpaca-delete` all of the above mentioned packages (magit, forge, ghub, transient, with-editor) *and* git-commit.
2. `M-x elpaca-update-menus` the `elpaca-menu-melpa` menu.
3. `M-x restart-emacs`

Elpaca should inherit from the updated recipes and install the packages based off those when Emacs is restarted.

**Comment by srcreigh** [comment]
for others coming here via google, a working method from #350 to install Magit via Elpaca is this:

```
(elpaca transient)
(elpaca (magit :wait t))
```

the simple `(elpaca magit)` does not work for me despite my best efforts to use latest everything.

Thanks progfolio

**Comment by cxwx** [comment]
Does the problem solved now？

**Comment by aarvay** [comment]
**For `elpaca-use-package-mode` users:**

Declare `transient` before any packages that depend on it (like `magit`, `forge`, `gptel`):

```elisp
(use-package transient
  :ensure (:wait t))

(use-package magit
  ;; ... your config
  )

(use-package forge
  :after magit)
